CS2-SimpleAdmin.cs

```diff
@@ -916,18 +916,28 @@ public void OnRcomCommand(CCSPlayerController? caller, CommandInfo command)
 		Logger.LogInformation($"{caller!.PlayerName} executed command ({command.ArgString}).");
 	}
 
+
+    string[] BannedGiveItems = {"weapon_knife_canis","weapon_falchion","weapon_flip","weapon_knife_outdoor","weapon_knife_widowmaker","weapon_elite","weapon_knife_skeleton",
+	"weapon_knife_push","weapon_knife_gypsy_jackknife","weapon_gut","weapon_tactical","weapon_knife_ursus","weapon_knife_stiletto","weapon_bayonet","weapon_m9_bayonet",
+	"weapon_karambit","weapon_butterfly","weapon_survival_bowie","weapon_knife_cord"};
     [ConsoleCommand("css_give")]
 	[RequiresPermissions("@css/give")]
 	[CommandHelper(minArgs: 2, usage: "<#UserId Or Name> <WeaponName>", whoCanExecute: CommandUsage.CLIENT_AND_SERVER)]
 	public void OnGiveCommand(CCSPlayerController? caller, CommandInfo command)
 	{
-
 		if(!GetTarget(command, out var player) || player == null || !player.IsValid) return;
+		string weaponName = command.GetArg(2);
 
-		string weapon = command.GetArg(2);
-		player.GiveNamedItem(weapon);
-
-		Server.PrintToChatAll(Helper.ReplaceTags($" {Config.Prefix} {Config.Messages.AdminGiveMessage}".Replace("{ADMIN}", caller?.PlayerName == null ? "Console" : caller.PlayerName).Replace("{WEAPON}", weapon.Split("_")[1])).Replace("{PLAYER}",player.PlayerName));
+		//check if weapon is inside the bannedItems
+		foreach(string bannedWeapon in BannedGiveItems){
```

Comment:
Just check if weapon contains `knife_` or `bagnet`

Reply:
weapon_falchion
weapon_flip
are examples of knifes that don't have 'knife' in their names

Reply:
> weapon_falchion weapon_flip are examples of knifes that don't have 'knife' in their names

All valid knives class
```		{ "weapon_knife", "Default Knife" },
		{ "weapon_knife_m9_bayonet", "M9 Bayonet" },
		{ "weapon_knife_karambit", "Karambit" },
		{ "weapon_bayonet", "Bayonet" },
		{ "weapon_knife_survival_bowie", "Bowie Knife" },
		{ "weapon_knife_butterfly", "Butterfly Knife" },
		{ "weapon_knife_falchion", "Falchion Knife" },
		{ "weapon_knife_flip", "Flip Knife" },
		{ "weapon_knife_gut", "Gut Knife" },
		{ "weapon_knife_tactical", "Huntsman Knife" },
		{ "weapon_knife_push", "Shadow Daggers" },
		{ "weapon_knife_gypsy_jackknife", "Navaja Knife" },
		{ "weapon_knife_stiletto", "Stiletto Knife" },
		{ "weapon_knife_widowmaker", "Talon Knife" },
		{ "weapon_knife_ursus", "Ursus Knife" },
		{ "weapon_knife_css", "Classic Knife" },
		{ "weapon_knife_cord", "Paracord Knife" },
		{ "weapon_knife_canis", "Survival Knife" },
		{ "weapon_knife_outdoor", "Nomad Knife" },
		{ "weapon_knife_skeleton", "Skeleton Knife" }
```

Reply:
I don't know any better solution rather than checking the full names and valve may add more knifes that don't contain 'knife' in their ids

Reply:
> Just check if weapon contains `knife_` or `bagnet`

.

Reply:
let me check 1sec

Reply:
yes my bad I was looking at somewhere else for the ids 😞

Reply:
You did some unnecessary work, but at least you know now :D

Reply:
Still wrong, u need to check for `_knife` to not block default `weapon_knife`